Simplified/Reader2/ReaderStackConfiguration/AdobeDRM/NYPLAdobeContentProtectionService.swift

```diff
@@ -0,0 +1,51 @@
+//
+//  NYPLAdobeContentProtectionService.swift
+//  Simplified
+//
+//  Created by Ettore Pasquini on 5/11/21.
+//  Copyright © 2021 NYPL Labs. All rights reserved.
+//
+
+#if FEATURE_DRM_CONNECTOR
+
+import Foundation
+import R2Shared
+import R2Streamer
+import R2Navigator
+
+final class NYPLAdobeContentProtectionService: ContentProtectionService {
+  var error: Error?
+  let context: PublicationServiceContext
+
+  init(context: PublicationServiceContext) {
+    self.context = context
+    self.error = nil
+    if let adobeFetcher = context.fetcher as? AdobeDRMFetcher {
+      if let drmError = adobeFetcher.container.epubDecodingError {
+        self.error = NSError(domain: "Adobe DRM decoding error",
+                             code: NYPLErrorCode.adobeDRMFulfillmentFail.rawValue,
+                             userInfo: [
+                              "AdobeDRMContainer error msg": drmError
+                             ])
+      }
+    }
+  }
+
+  /// A restricted publication has a limited access to its manifest and
+  /// resources and can’t be rendered with a Navigator. It is usually
+  /// only used to import a publication to the user’s bookshelf.
+  var isRestricted: Bool {
+    context.publication.ref == nil || error != nil
+  }
+
+  var rights: UserRights {
+    isRestricted ? AllRestrictedUserRights() : UnrestrictedUserRights()
```

Comment:
This might be an issue for Adobe's certification, as the navigator will allow any text copy. I suggest always using `AllRestrictedUserRights()` until you implement your own `UserRights` for ACS. [It's implemented like this with LCP](https://github.com/readium/r2-lcp-swift/blob/b8adc686c392b3279f57259d6904af599b9737d3/readium-lcp-swift/License/License.swift#L67-L105).

Reply:
that's a good point... well my understanding was (please correct me I'm wrong, @killerpuppytails )  that we had those rights on SimplyE but Open eBooks has more restrictions  i.e. disabling the "Copy" / "Look up" pop up menu -- that currently we address elsewhere. I have to say I only recently found the more declaratively options for doing so inside R2, so I should file a ticket for improving that.

Reply:
You're absolutely correct, @ettore . Open eBooks has many more restrictions and we explicitly were asked to turn it off by partners. However, since we will be removing Adobe from OE for the next launch, we have more options we can explore. I'll set up a conversation to confirm.